File: 566-reshape-the-matrix/566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        
        x, y = len(mat), len(mat[0])
        if(x == 0 or r*c != x*y):
            return mat
        ans = [[0 for _ in range(c)] for _ in range(r)]
        q = deque()
        for i in range(x):
            for j in range(y):
                q.append(mat[i][j])
        
        for i in range(r):
            for j in range(c):
                ans[i][j] = q.popleft()
        return ans

File: 566-reshape-the-matrix/566-reshape-the-matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) 
    {
        int rows = mat.size(); int cols = mat[0].size();
        if(rows == 0 || rows * cols != r * c)
            return mat;
        deque <int> q;
        for(int i = 0; i < rows; i++)
            for(int j = 0; j < cols; j++)
                q.push_back(mat[i][j]);
        vector <vector<int>> ans(r, vector <int>(c, 0));
        for(int i = 0; i < r; i++)
            for(int j = 0; j < c; j++)
            {
                ans[i][j] = q.front();
                q.pop_front();
            }
        
        return ans;
        
    }
};